charts/cortex/templates/ingester/ingester-statefulset.yaml

```diff
@@ -141,3 +142,163 @@ spec:
+    rollout-group: ingester
+  annotations:
+    {{- toYaml $.Values.ingester.annotations | nindent 4 }}
+spec:
+  {{- if not $.Values.ingester.autoscaling.enabled }}
+  replicas: {{ $.Values.ingester.replicas }}
+  {{- end }}
+  selector:
+    matchLabels: {{ include "cortex-ingester.matchLabels" $ | nindent 6 }}
+{{ $.Values.topologyAware.zoneLabel | nindent 6 }}: {{ $availabilityCell }}
+      rollout-group: ingester
+  updateStrategy:
+    {{- toYaml $.Values.ingester.statefulStrategy | nindent 4 }}          
+  podManagementPolicy: "{{ $.Values.ingester.statefulSet.podManagementPolicy }}"
+  serviceName: {{ template "cortex-ingester.headlessname-zone-aware" dict "Values" $.Values "Release" $.Release "Chart" $.Chart "AvailabilityCell" $availabilityCell }}
+  {{- if $.Values.ingester.persistentVolume.enabled }}
+  volumeClaimTemplates:
+    - metadata:
+        name: storage
+        {{- if $.Values.ingester.persistentVolume.annotations }}
+        annotations:
+          {{ toYaml $.Values.ingester.persistentVolume.annotations | nindent 10 }}
+        {{- end }}
+      spec:
+        {{- if $.Values.ingester.persistentVolume.storageClass }}
+        {{- if (eq "-" $.Values.ingester.persistentVolume.storageClass) }}
+        storageClassName: ""
+        {{- else }}
+        storageClassName: "{{ $.Values.ingester.persistentVolume.storageClass }}"
+        {{- end }}
+        {{- end }}
+        accessModes:
+          {{ toYaml $.Values.ingester.persistentVolume.accessModes | nindent 10 }}
+        resources:
+          requests:
+            storage: "{{ $.Values.ingester.persistentVolume.size }}"
+  {{- end }}
+  template:
+    metadata:
+      labels: {{ include "cortex-ingester.labels" $ | nindent 8 }}
+{{ $.Values.topologyAware.zoneLabel | nindent 8 }}: {{ $availabilityCell }}
+        rollout-group: ingester
+{{- with $.Values.ingester.podLabels }}
+{{- toYaml . | indent 8 }}
+{{- end }}
+      annotations:
+        checksum/config: {{ include "cortex.configChecksum" $ }}
+        {{- with $.Values.ingester.podAnnotations }}
+        {{- toYaml . | nindent 8 }}
+        {{- end }}
+    spec:
+      serviceAccountName: {{ $.Values.ingester.serviceAccount.name | default (include "cortex.serviceAccountName" $ ) }}
+      {{- if $.Values.ingester.priorityClassName }}
+      priorityClassName: {{ $.Values.ingester.priorityClassName }}
+      {{- end }}
+      securityContext: {{- toYaml $.Values.ingester.securityContext | nindent 8 }}
+      initContainers:
+        {{- toYaml $.Values.ingester.initContainers | nindent 8 }}
+      {{- if $.Values.image.pullSecrets }}
+      imagePullSecrets:
+      {{- range $.Values.image.pullSecrets }}
+        - name: {{ . }}
+      {{- end }}
+      {{- end }}
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: {{ $.Values.topologyAware.zoneLabel }}
```

Comment:
Are we sure that the zone label on the nodes is exactly the same as the one on the statefulsets? I think I would I chose the label name for the statefulset and let the user chose which label to match on the nodes. Minor.